hs_composite_resource/tests/test_composite_resource_auto_aggregate.py

```diff
@@ -0,0 +1,183 @@
+
+    def test_auto_aggregate_file_add_tif(self):
+        """test that auto-aggregate works on tif file add"""
+
+        self.create_composite_resource()
+
+        self.assertEqual(0, GeoRasterLogicalFile.objects.count())
+
+        # test add a file that auto-aggregates
+        open_file = open(self.test_file_path.format("small_logan.tif"), 'r')
+        resource_file_add_process(resource=self.composite_resource,
+                                  files=(open_file,), user=self.user)
+
+        self.assertEqual(1, GeoRasterLogicalFile.objects.count())
+
+    def test_auto_aggregate_file_add_tiff(self):
+        """test that auto-aggregate works on tiff file add"""
+
+        self.create_composite_resource()
+
+        self.assertEqual(0, GeoRasterLogicalFile.objects.count())
+
+        # test add a file that auto-aggregates
+        open_file = open(self.test_file_path.format("small_logan.tiff"), 'r')
+        resource_file_add_process(resource=self.composite_resource,
+                                  files=(open_file,), user=self.user)
+
+        self.assertEqual(1, GeoRasterLogicalFile.objects.count())
+
+    def test_auto_aggregate_file_add_sqlite(self):
+        """test that auto-aggregate works on sqlite file add"""
+
+        self.create_composite_resource()
+
+        self.assertEqual(0, TimeSeriesLogicalFile.objects.count())
+
+        # test add a file that auto-aggregates
+        open_file = open(self.test_file_path.format("ODM2.sqlite"), 'r')
+        resource_file_add_process(resource=self.composite_resource,
+                                  files=(open_file,), user=self.user)
+
+        self.assertEqual(1, TimeSeriesLogicalFile.objects.count())
+
+    def test_auto_aggregate_file_add_refts(self):
+        """test that auto-aggregate works on refts file add"""
+
+        self.create_composite_resource()
+
+        self.assertEqual(0, RefTimeseriesLogicalFile.objects.count())
+        # test add a file that auto-aggregates
+        open_file = open(self.test_file_path.format("multi_sites_formatted_version1.0.refts.json"),
+                         'r')
+        resource_file_add_process(resource=self.composite_resource,
+                                  files=(open_file,), user=self.user)
+
+        self.assertEqual(1, RefTimeseriesLogicalFile.objects.count())
+
+    def test_auto_aggregate_file_add_nc(self):
+        """test that auto-aggregate works on nc file add"""
+
+        self.create_composite_resource()
+
+        self.assertEqual(0, NetCDFLogicalFile.objects.count())
+        # test add a file that auto-aggregates
+        open_file = open(self.test_file_path.format("netcdf_valid.nc"), 'r')
+        resource_file_add_process(resource=self.composite_resource,
+                                  files=(open_file,), user=self.user)
+        # because of auto aggregation, there should be 2 files
+
+        self.assertEqual(1, NetCDFLogicalFile.objects.count())
+
+    def test_auto_aggregate_file_add_geo_feature(self):
```

Comment:
I don't see unit tests that cover auto aggregation creation when a file is added to a folder.

Reply:
thanks for catching that.  I've added the tests now.